Entity 및 Repository 정의

hyntato opened the pull request:
### What
- 기본 Entity, Repository 정의
- Workbench에서 ERD를 생성하여 Table, Relationship 확인

### ERD
<img width="787" alt="스크린샷 2022-09-28 오후 10 10 56" src="https://user-images.githubusercontent.com/56067179/192788976-3c9de5b4-eb4c-43ae-a69e-b363c5880b49.png">

hyntato committed:
config: database 설정

- database 설정
- 테스트용 스키마 및 데이터 파일 생성

hyntato committed:
entity 정의

- User, Post, Comment, LikePost, LikeComment, Hit, HitSession 정의
- BaseEntity 정의: entity 생성, 수정 시간 Auditing 기능 수행

hyntato committed:
repository 정의

- entity에 대한 기본 repository 생성

hyntato committed:
repository 테스트 코드

sungyujeon commented on blog/src/main/java/com/hyntato/blog/entity/Hit.java:
```diff
@@ -0,0 +1,25 @@
+    @Column(name = "hit_id")
+    Long id;
+
+    @OneToOne
```
ManyToOne, OneToOne은 FetchType이 default로 eager인데, Hit 조회 시 Post가 자동으로 fetch 될 것 같습니다. Hit 조회 시 Post 데이터를 전부 fetch 하는 것이 효율적일지 고민해보면 좋을 것 같습니다

hyntato replied:
@sungyujeon 피드백 반영했습니다! 짚어주셔서 감사합니다~!

sungyujeon commented on blog/src/main/java/com/hyntato/blog/entity/Hit.java:
```diff
@@ -0,0 +1,25 @@
+
+@Entity
+@Getter
+@Setter
```
AllArgsConstructor annotation을 선언해주었는데, 추가적으로 setter를 모두 생성하는 것은 불필요하지 않을지 생각해보면 좋을 것 같습니다.

hyntato replied:
@sungyujeon builder를 사용하는 방법으로 바꿔봤습니다 :)

sungyujeon commented:
추가적으로 schema와 기본 data sql을 resources에 추가해둔 점이 좋은 것 같습니다.

hyntato committed:
add: OneToOne FetchType 추가

hyntato committed:
add: Builder 추가, Setter 삭제

hyntato committed:
refactor: 테스트 코드 수정

- 테스트 코드 종속 관계 제거
- @Transactional 추가: 테스트 코드 실행 시, 자동 롤백 이루어지지 않는 문제 해결
- 추가적으로, 테스트 코드에서 User, Post, Comment를 생성하는 부분이 반복되는 것을 해결해 볼 예정입니다. 추후 리팩토링을 위해 반복되는 부분을 그대로 커밋하오니, 코드가 길더라도 양해 부탁드립니다..!

hyntato committed:
refactor: 테스트 코드 리팩토링

- @BeforeAll Annotation을 사용하여 테스트 이전에 필요한 데이터를 한 번만 생성하도록 리팩토링 했습니다.

hyntato committed:
remove: hit, likeComment, likePost 테스트 코드 삭제
